policies/testing.md

```diff
@@ -0,0 +1,72 @@
+Testing Policy
+==============
+
+This applies to all stable and development branches of the main code repository.
+
+Within this policy "functional behaviour" means what the system does, i.e.
+given a set of inputs it will produce a set of outputs. This does not include
+how the system does it. For example refactoring or performance improvements do
+not affect functional behaviour.
+
+Except where noted below:
+
+- All Pull Requests adding new functionality to the applications, libraries,
+providers or engines must include suitable tests.
+- All Pull Requests fixing a functional behavior defect in the applications,
+libraries, providers or engines must include a test for that defect.
+
+Pull Requests that do not change the functional behaviour of the applications,
+libraries, providers or engines do not require tests to be added. For example
+the following types of changes do not require tests:
+ - Changes to comments, formatting, internal naming or similar
+ - Fixes for performance
+ - Refactoring
+
+To be explicit the above statement means that tests are not required for changes
+in:
+ - Documentation (including CHANGES.md/NEWS.md)
+ - The test suite
+ - perl utilities
+ - Include files
+ - Build system
+ - Demos
+
+Pull Requests that only affect minor cosmetic output do not require tests.
+
+By agreement of the Pull Request reviewers it is acceptable for the tests to be
+added via a different Pull Request to the main Pull Request.
```

Comment:
Do we want to enforce that it gets written? And so that the pull requests that makes the behavioural changes can only get merged as the same time as the one testing it?

Reply:
I'm not sure. The current text doesn't require it. I'd like to hear the OTC view on this.

Reply:
**OTC does not see the need to change the policy in this way for now.**